File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        ans = [0] * n
        for i in range(n):
            k = -1
            for j in range(i + 1, n):
                k = max(k, arr[j])
            ans[i] = k
        return ans

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0, k = prices[0], n = prices.size();
        for(int i = 0; i < n; i++) {
            ans = max(ans, prices[i] - k);
            k = min(k, prices[i]);
        }
        return ans;
    }
};
